### DISCUSSION TITLE
How to use pyvistaqt to run plotter in separate thread

### DISCUSSION THREAD

**Discussion by royvelich** [pyvista/pyvista#4675] [Q&A]
Hi,
I probably don't understand how to use the `BackgroundPlotter` properly, I would love to know what am I doing wrong.
So, I want that my pyvista window/plotter will run in a separate thread, while my main thread is doing some calculations and updates the plotter's scene continuously (the main thread is training a neural network in my case, I use the plotter to visualize its results).
So, I came up with the following template, just for experimentation:
```python
import os
os.environ["QT_API"] = "pyqt5"
import pyvista as pv
from pyvistaqt import BackgroundPlotter
from threading import Thread
from time import sleep

def update_mesh():
    plotter = BackgroundPlotter(shape=(1, 2))
    sphere = pv.Sphere()
    plotter.add_mesh(sphere)
    plotter.app.exec_()

t = Thread(target=update_mesh, args=())
t.start()

while True:
    sleep(2)
    print('Hi')
```

 At this point, I do not understand how can I interact with the plotter object through the main thread. If I try to create the `BackgroundPlotter` at the main thread and pass it into the worker thread as an argument, the code crashes.

What is the correct way to do it?

Thanks

**Comment by banesullivan** [comment]
There are a few other issues/discussions floating around on this topic. In short, you can't. the TLDR is that VTK has run in the main event thread

Reference:

- https://github.com/pyvista/pyvista/discussions/4006
- https://github.com/Kitware/trame/discussions/173#discussioncomment-4605366
